### ISSUE TITLE
[SFT] Chat template may silently drop `tools` field for tool-use / agentic training data

### ISSUE THREAD

**Issue by yaoyu-33** [NVIDIA-NeMo/Megatron-Bridge#4072]
## Summary

When preparing tool-use / agentic SFT data, the chat template application path in Megatron-Bridge can silently drop the `tools` argument that should be passed into `tokenizer.apply_chat_template(...)`. Conversations whose turns depend on tool schemas (function calling, MCP-style tools, agentic data in general) are then tokenized without their tool definitions, producing training targets that look syntactically valid but are semantically wrong.

This is a silent failure: no exception is raised, loss curves look reasonable, and the bug only shows up as degraded tool-use behavior in the trained model.

## Context

HF's `apply_chat_template` requires an explicit `tools=[...]` kwarg for tool-aware Jinja templates — omitting it silently emits the no-tools branch of the template. If the dataset's `tools` field is not threaded through to that call, the loss is computed against a prompt that the model would never see at inference time.

## Repro sketch

1. Take an SFT sample with a `tools` field (OpenAI-style tool schema list) and at least one assistant turn that calls a tool.
2. Run it through the current SFT data path (dataset → chat template → tokenization).
3. Compare the resulting `input_ids` to a direct call:
   ```python
   tokenizer.apply_chat_template(
       messages, tools=tools, add_generation_prompt=False, tokenize=True
   )
   ```
4. Expected: identical token sequence, including the rendered tool schema block.
   Actual: Bridge output is missing the tool schema region (template falls through to its `tools is none` branch).

## Proposed fix

- Audit every call site of `apply_chat_template` in `src/megatron/bridge/data/` and ensure `tools=` is threaded through when present on the sample.
- Add a dataset-side schema check: if a sample contains a `tools` field and the tokenizer's chat template references `tools`, fail loudly when `tools` would be dropped before tokenization.
- Add a unit test using a tool-aware Jinja chat template that asserts the rendered token sequence contains the tool schema markers.
- Document the expected sample schema for agentic SFT (`messages`, `tools`, and any per-turn `tool_calls` / `tool_call_id`).

## Acceptance criteria

- [ ] All SFT data paths forward `tools` to `apply_chat_template` when present on the sample.
- [ ] Unit test fails on `main` today and passes after the fix.
- [ ] Loud error (not silent drop) when a sample's `tools` field cannot be honored by the active tokenizer template.
